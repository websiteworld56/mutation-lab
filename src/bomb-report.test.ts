import { table } from 'table';
import BombReport from "./bomb-report";

jest.mock('table');

describe('Bomb Report', () => {
    it('should prepare document correctly', async () => {
        // ARRANGE
        (table as any).mockReturnValueOnce('perfect table');
        const thread1 = {
            originalStartLine: 1,
            originalLine: 2,
            comments: {
                nodes: [
                    {
                        path: 'some/path/to/source-file.js'
                    }
                ]
            }
        };
        const thread2 = {
            line: 3,
            originalLine: 3,
            comments: {
                nodes: [
                    {
                        path: 'some/path/to/another-source-file.js'
                    }
                ]
            }
        };
        const report = new BombReport();
        report.pushBombOutcome(thread1, false);
        report.pushBombOutcome(thread2, true);

        // ACT
        const actual = report.generateReport();

        // ASSERT
        expect(actual).toStrictEqual('perfect table');
        expect((table as any).mock.calls).toMatchSnapshot();
    });

    it('should return an empty string if no outcomes where captured', async () => {
        const report = new BombReport();
        expect(report.generateReport()).toStrictEqual('');
    });
});
