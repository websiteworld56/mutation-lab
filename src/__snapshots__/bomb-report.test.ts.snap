// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bomb Report should prepare document correctly 1`] = `
Array [
  Array [
    Array [
      Array [
        "File Name",
        "Line Numbers",
        "Bomb Defused",
      ],
      Array [
        "source-file.js",
        "1-2",
        "no",
      ],
      Array [
        "another-source-file.js",
        3,
        "yes",
      ],
    ],
  ],
]
`;
